concepts/70 Data Binding/00 Specify a Data Source/60 Custom Data Sources/2 Load Data/5 Server-Side Data Processing.md

```diff
@@ -75,70 +75,79 @@ The following example shows a **CustomStore** that sends data processing setting
     import CustomStore from 'devextreme/data/custom_store';
+    import { lastValueFrom } from 'rxjs';
+    import DataSource from 'devextreme/data/data_source';
 
     @Component({
         selector: 'app-root',
         templateUrl: './app.component.html',
         styleUrls: ['./app.component.css']
     })
+
     export class AppComponent {
-        customDataSource: CustomStore;
+        customDataSource: DataSource;
         constructor(private http: HttpClient) {
-            const isNotEmpty = (value) => value !== undefined && value !== null && value !== '';
-            
-            this.customDataSource = new CustomStore({
-                key: 'ID',
-                load: (loadOptions) => {
-                    let params: HttpParams = new HttpParams();
-
-                    [
-                        'filter',
-                        'group', 
-                        'groupSummary',
-                        'parentIds',
-                        'requireGroupCount',
-                        'requireTotalCount',
-                        'searchExpr',
-                        'searchOperation',
-                        'searchValue',
-                        'select',
-                        'sort',
-                        'skip',     
-                        'take',
-                        'totalSummary', 
-                        'userData'
-                    ].forEach(function(i) {
-                        if(i in loadOptions && isNotEmpty(loadOptions[i])) {
-                            params[i] = JSON.stringify(loadOptions[i]);
-                        }
-                    });
-
-                    return this.http.get('https://mydomain.com/MyDataService', { params: params })
-                        .toPromise()
-                        .then(response => {
+            const isNotEmpty = (value: unknown) => value !== undefined && value !== null && value !== '';
+
+            this.customDataSource = new DataSource({
+                store: new CustomStore({
+                    key: 'ID',
+                    load: (loadOptions: LoadOptions) => {
+                        let params: HttpParams = new HttpParams();
+
+                        [
+                            'filter',
+                            'group',
+                            'groupSummary',
+                            'parentIds',
+                            'requireGroupCount',
+                            'requireTotalCount',
+                            'searchExpr',
+                            'searchOperation',
+                            'searchValue',
+                            'select',
+                            'sort',
+                            'skip',
+                            'take',
+                            'totalSummary',
+                            'userData',
+                        ].forEach(function (i) {
+                            if (i in loadOptions && isNotEmpty(loadOptions[i as keyof LoadOptions])) {
```

Comment:
Small refactoring:

            const optionValue = loadOptions[i as keyof LoadOptions];
            if (i in loadOptions && isNotEmpty(optionValue)) {
              params = params.set(i, JSON.stringify(optionValue));
            }

Reply:
Done!